pkg/services/hivelocity/device/device.go

```diff
@@ -425,11 +426,20 @@ func (s *Service) actionDeviceProvisioned(ctx context.Context) actionResult {
 }
 
 func (s *Service) verifyAssociatedDevice(device *hv.BareMetalDevice) error {
-	if !s.scope.HivelocityCluster.DeviceTag().IsInStringList(device.Tags) {
-		return errClusterTagNotFound
+	deviceTag, err := hvtag.ClusterTagFromList(device.Tags)
+	if err != nil {
+		return err
+	}
+	if s.scope.HivelocityCluster.Name != deviceTag.Value {
+		return fmt.Errorf("expected %q got %q: %w", s.scope.HivelocityCluster.Name, deviceTag.Value, errNoDeviceAvailable)
```

Comment:
I think this error message shouldn't be errNoDeviceAvailable. 
It should be "errWrongClusterTag"

Reply:
fixed.